plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1930,7 +1935,8 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
       return start_block_result::failed;
    }
 
-   block_num_type             head_block_num    = chain.head().block_num();
+   block_handle               head              = chain.head();
+   block_num_type             head_block_num    = head.block_num();
```

Comment:
detail, I think it would be nicer to define `head` in the if condition where it is used, instead of so much earlier.

Reply:
I was thinking moving just `block_handle head = chain.head();`, but never mind that is not nicer.